pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/BaseMultipleSegmentsConversionExecutor.java

```diff
@@ -352,6 +336,59 @@ public List<SegmentConversionResult> executeTask(PinotTaskConfig pinotTaskConfig
     }
   }
 
+  private int parallelDownloadAndUntarSegments(String[] downloadURLs, String[] segmentNames, String tableNameWithType,
```

Comment:
I think better for this method to return List<SegmentMetadataImpl> as thats more intuitive for a method called "parallelDownloadAndUntarSegments". and the caller can iterate and populate numRecords. 

this will also help in creating the same order of inputSegmentDirs as "segmentNames" as we can use SegmentMetadataImpl.getName() and sort based on index of segmentNames for that segment name